app/services/precheck_eligibility_service.rb

```diff
@@ -0,0 +1,44 @@
+class PrecheckEligibilityService < ApplicationService
+  attr_accessor :family
+
+  def call
+    pending_approval? &&
+      within_time_window? &&
+      children_approved? &&
+      chargeable_staff_number? &&
+      housing_preference_confirmed?
+  end
+
+  private
+
+  def_delegator :family, :pending_approval?
+  def_delegator :family, :attendees
+  def_delegator :family, :children
+  def_delegator :family, :chargeable_staff_number?
+  def_delegator :family, :housing_preference
+
+  def within_time_window?
+    return false if earliest_attendee_arrival_date.blank?
+
+    earliest_precheck_date = earliest_attendee_arrival_date - 10.days
+    latest_precheck_date   = earliest_attendee_arrival_date - 2.days
+
+    (earliest_precheck_date..latest_precheck_date).cover? Time.zone.now
```

Comment:
This means that there is only an 8 day window to do the precheck? 🤔

Reply:
Yeah that's correct, this is what Cru requested but now that you bring it up I agree that it does feel like a pretty short window. I added it as a note to discuss with them more!